src/request.ts

```diff
@@ -77,27 +77,59 @@ export async function buildRequestUrl(
 }
 
 /**
- * The error type returned by `executeRequestError`.
+ * The error type returned by the SDK whenever it makes a network request
+ * (internally, this happens in `executeRequest`). It implements, so is
+ * compatible with, `AxiosError`.
  */
-export class ExecuteRequestError extends Error {
-  originalError: AxiosError;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class ExecuteRequestError<T = any, D = any> extends Error implements AxiosError {
   responseStatus: number | null;
   responseMessage: string | null;
 
+  // Properties required by `AxiosError`.
+
+  config: AxiosRequestConfig<D>;
+  code?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request?: any;
+  response?: AxiosResponse<T, D>;
+  isAxiosError: boolean;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  toJSON: () => object;
+
   /**
    * Creates an `ExecuteRequestError`.
    *
    * @param message - The error message.
-   * @param axiosError - The original axios error.
+   * @param axiosError - The original Axios error.
    * @param responseStatus - The response status, if found in the original error.
    * @param responseMessage - The response message, if found in the original error.
    */
   constructor(message: string, axiosError: AxiosError, responseStatus: number | null, responseMessage: string | null) {
+    // Include this check since ExecuteRequestError implements AxiosError, but
+    // we only expect original errors from Axios here.
+    if (axiosError instanceof ExecuteRequestError) {
+      throw new Error(
+        "Could not instantiate an `ExecuteRequestError` from an `ExecuteRequestError`, an original error from axios was expected"
+      );
+    }
+
+    // Set `Error` fields.
     super(message);
-    this.originalError = axiosError;
+    this.name = "ExecuteRequestError";
+
+    // Set `ExecuteRequestError` fields.
     this.responseStatus = responseStatus;
     this.responseMessage = responseMessage;
 
+    // Set properties required by `AxiosError`.
+    this.config = axiosError.config;
```

Comment:
wondering if you could do `Object.assign(this, axiosError)`?

Reply:
So I tried to do 

```ts
// Set properties required by `AxiosError`.
delete axiosError["message"];
delete axiosError.name;
Object.assign(this, axiosError);
```

But I was getting an error due to deleting non-optional fields in `AxiosError`. So I can copy over `axiosError` into a new object and then delete the fields but then all this seems really hacky.

Reply:
I tried this

```ts
// Set properties required by `AxiosError`.
Object.defineProperty(this, "message", { configurable: false, writable: false });
Object.defineProperty(this, "name", { configurable: false, writable: false });
Object.assign(this, axiosError);
```

But now I'm getting an error about 

```ts
config: AxiosRequestConfig<D>;
```

and etc. not being definitely defined in the constructor. 😅 Oh well, that was a fun and informative 10-minute diversion.

Reply:
haha ok thanks for entertaining the suggestion